core/src/main/java/io/grpc/Channel.java

```diff
@@ -59,4 +63,18 @@
    */
   public abstract <RequestT, ResponseT> ClientCall<RequestT, ResponseT> newCall(
       MethodDescriptor<RequestT, ResponseT> methodDescriptor, CallOptions callOptions);
+
+  /**
+   * Pings the remote endpoint to verify that the transport is still active. When an acknowledgement
+   * is received, the given callback will be invoked using the given executor.
+   *
+   * <p>If the underlying transport has no mechanism by when to send a ping, this method may throw
+   * an {@link UnsupportedOperationException}. The operation may {@linkplain
+   * io.grpc.internal.ClientTransport.PingCallback#pingFailed(Throwable) fail} due to transient
+   * transport errors. In that case, trying again may succeed.
+   *
+   * @see ClientTransport#ping(ClientTransport.PingCallback, Executor)
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/737")
+  public abstract void ping(final ClientTransport.PingCallback callback, final Executor executor);
```

Comment:
Can we not add this to Channel? I'm still hoping it can be removed and just be exposed to load balancers.